packages/typespec-test/test/batch_modular/generated/typespec-ts/src/models/pagingTypes.ts

```diff
@@ -0,0 +1,36 @@
+// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT license.
+
+/**
+ * An interface that tracks the settings for paged iteration
+ */
+export interface PageSettings {
```

Comment:
Is this file the same across all services?  How about moving this into the common library?

Reply:
Currently we have same interface name in paging-core https://github.com/Azure/azure-sdk-for-js/blob/main/sdk/core/core-paging/src/models.ts#L7. But we removed the `maxpagesize` considering it is not supported(see [dicusssion here](https://github.com/Azure/autorest.typescript/issues/1856#issuecomment-1711832806)).

We plan to add them to core when this version is stable in future.

Reply:
Yeah, we could consider doing an update to remove `maxPageSize` and make any other needed breaking adjustments in the new core and align to that -- @deyaaeldeen were there other changes we had thought about? I dimly recall us having regrets over the generic parameters?

Reply:
Yes we can make the generic page settings extend from the default page settings type, see your comment in https://github.com/Azure/azure-sdk-for-js/pull/23404#discussion_r990535750. Otherwise, I created this wish list back in the day which has one item for now which will simplify the implementation a bit: https://github.com/Azure/azure-sdk-for-js/issues/21836

Reply:
I created an issue in sdk repo to focus on the alignment https://github.com/Azure/azure-sdk-for-js/issues/27868.